src/export/ExportFFmpeg.cpp

```diff
@@ -665,27 +748,28 @@ bool ExportFFmpeg::InitCodecs(AudacityProject *project)
 }
 
 // Returns 0 if no more output, 1 if more output, negative if error
-static int encode_audio(AVCodecContext *avctx, AVPacket *pkt, int16_t *audio_samples, int nb_samples)
+static int encode_audio(const FFmpegFunctions& ffmpeg, AVCodecContextWrapper*avctx, AVPacketWrapper *pkt, int16_t *audio_samples, int nb_samples)
 {
    // Assume *pkt is already initialized.
 
    int i, ch, buffer_size, ret, got_output = 0;
-   AVMallocHolder<uint8_t> samples;
-   AVFrameHolder frame;
+   AVDataBuffer<uint8_t> samples;
+
+   std::unique_ptr<AVFrameWrapper> frame;
 
    if (audio_samples) {
-      frame.reset(av_frame_alloc());
+      frame = ffmpeg.CreateAVFrameWrapper();
+
       if (!frame)
-         return AVERROR(ENOMEM);
+         return -ENOMEM;
```

Comment:
Why is the error code negated?  AVERROR was a no-op macro, it didn't negate.

I wondered if this return value is passed to ffmpeg because this is a callback function?  But further examination tells me no.

Reply:
Hmm, yep, thats a type

Reply:
This is a bug in the Python generator (well, not a bug, it is ignorant about some libc constants). I will fix it in FFmpegTypes.h